thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/api/RcaInvestigationApi.java

```diff
@@ -114,4 +115,13 @@ public RcaInvestigationApi setUpdatedBy(final UserApi updatedBy) {
     this.updatedBy = updatedBy;
     return this;
   }
+
+  public AuthorizationConfigurationApi getAuth() {
```

Comment:
seems like auth is available for all crud entities? Should we place it a level above then?

Reply:
The crudApi's implement an interface (unlike the Dto's which extend an abstract class), so I don't think I can move this a level up.

Reply:
We can have getAuth and setAuth just like we have getId and setId in ThirdEyeCrudApi. However, not sure if we are there yet for all the entities. What entities are we not handling today? Else, setting the auth part can be there in CrudResource instead.

Reply:
Let's tackle this later though. don't see this as a blocker for this PR